### DISCUSSION TITLE
Finding the 100% for the graph and the data in JSON input

### DISCUSSION THREAD

**Discussion by JackKuiper** [grafana/grafana-infinity-datasource#300] [Q&A] [answered]
I'm kind of at a loss here..

further below is the JSON output I get
The table in Grafana starts withe the UDID part
I wanted to generate a pie chart wich should give me the iPads, MacBook, Apple TV... in that location.
the "count" down below" is the 100% and I need the count of model.type.

But is seriously have no clue as how to go about that and if infinity can help me with that.

thanks for any light one can shed on this.

thx,

Jack
 
{
    "code": 200,
    "count": 7,
    "devices": [
        {
            "UDID": "f5c7b5d74c099d02fea87a22d69cede89e3aeb6b",
            "locationId": 2,
            "serialNumber": "F9FTG97EHLF9",
            "assetTag": "ITaS-Labor-IP-0001",
            "class": "ipad",
            "model": {
                "name": "iPad (5th generation, Wi‑Fi)",
                "identifier": "iPad6,11",
                "type": "iPad"
            },
            "os": {
                "prefix": "iOS",
                "version": "15.4.0"
            },
            "name": "ITaS-Labor-IP-0001",
            "owner": {
                "id": 1026,
                "locationId": 2,
                "username": "xyz",
                "email": "xyz@ge.de",
                "firstName": "xy",
                "lastName": "z",
                "name": "xyz",
                "vpp": [],
                "notes": "",
                "modified": "2022-03-17 11:04:26"
            },
            "isManaged": true,
            "isSupervised": true,
            "enrollType": "dep",
            "depProfile": "ITaS Labor iOS",
            "groups": [
                "VPP Apps all iOS",
                "Local WiFi all Devices",
                "Local Restrictions all iOS",
                "Global Restrictions iOS",
                "Global WiFi all SuS",
                "VPP Apps all iOS",
                "Local WiFi all Devices",
                "Local Restrictions all iOS",
                "Global Restrictions iOS",
                "Global WiFi all SuS",
                "VPP Apps all iOS",
                "Local WiFi all Devices",
                "Local Restrictions all iOS",
                "Global Restrictions iOS",
                "Global WiFi all SuS",
                "VPP Apps all iOS",
                "Local WiFi all Devices",
                "Local Restrictions all iOS",
                "Global Restrictions iOS",
                "Global WiFi all SuS",
                "VPP Apps all iOS",
                "Local WiFi all Devices",
                "Local Restrictions all iOS",
                "Global Restrictions iOS",
                "Global WiFi all SuS"
            ],
            "batteryLevel": 0.44,
            "totalCapacity": 32,
            "availableCapacity": 17.8746,
            "iCloudBackupEnabled": false,
            "iCloudBackupLatest": "0",
            "iTunesStoreLoggedIn": false,
            "region": {
                "string": ""
            },
            "notes": "",
            "lastCheckin": "2022-04-20 12:17:56",
            "modified": "2022-04-19 08:01:02",
            "networkInformation": {
                "IPAddress": "178.203.43.142",
                "isNetworkTethered": "1",
                "BluetoothMAC": "b4:9c:df:9b:17:9a",
                "WiFiMAC": "b4:9c:df:9b:17:99",
                "VoiceRoamingEnabled": "0",
                "DataRoamingEnabled": "0",
                "PersonalHotspotEnabled": "0"
            }
        },

**Comment by yesoreyeram** [comment, marked as answer]
## With UQL:

```kusto
parse-json
| scope "devices"
| summarize count() by "model.type"
```

<img width="1548" alt="image" src="https://user-images.githubusercontent.com/153843/164235405-716f8fcc-9a47-491e-830c-5c1bba8065e5.png">

## With JSON:

With JSON type, you need a group by transformation

<img width="1588" alt="image" src="https://user-images.githubusercontent.com/153843/164235704-37ea3eb4-3e10-4886-8b67-3ccbc92b6227.png">
<img width="1179" alt="image" src="https://user-images.githubusercontent.com/153843/164235735-fca99c6b-fd90-43e1-83ac-54e94a6868c8.png">

**Reply by JackKuiper** [reply]
Thanks a billion.
With UQL it seems a lot more powerful.
i will look into that and try to install that first
